salt/transport/tcp.py

```diff
@@ -192,187 +180,39 @@ def run(self):
-
-        # crypt defaults to 'aes'
-        self.crypt = kwargs.get("crypt", "aes")
-
-        self.io_loop = kwargs.get("io_loop") or salt.ext.tornado.ioloop.IOLoop.current()
-
-        if self.crypt != "clear":
-            self.auth = salt.crypt.AsyncAuth(self.opts, io_loop=self.io_loop)
-
-        resolver = kwargs.get("resolver")
-
-        parse = urllib.parse.urlparse(self.opts["master_uri"])
-        master_host, master_port = parse.netloc.rsplit(":", 1)
-        self.master_addr = (master_host, int(master_port))
-        self._closing = False
-        self.message_client = SaltMessageClientPool(
-            self.opts,
-            args=(
-                self.opts,
-                master_host,
-                int(master_port),
-            ),
-            kwargs={
-                "io_loop": self.io_loop,
-                "resolver": resolver,
-                "source_ip": self.opts.get("source_ip"),
-                "source_port": self.opts.get("source_ret_port"),
-            },
-        )
-
-    def close(self):
-        if self._closing:
-            return
-        log.debug("Closing %s instance", self.__class__.__name__)
-        self._closing = True
-        self.message_client.close()
-
-    # pylint: disable=W1701
-    def __del__(self):
-        try:
-            self.close()
-        except OSError as exc:
-            if exc.errno != errno.EBADF:
-                # If its not a bad file descriptor error, raise
-                raise
-
-    # pylint: enable=W1701
-
-    def _package_load(self, load):
-        return {
-            "enc": self.crypt,
-            "load": load,
-        }
+class Resolver:
 
-    @salt.ext.tornado.gen.coroutine
-    def crypted_transfer_decode_dictentry(
-        self, load, dictkey=None, tries=3, timeout=60
-    ):
-        if not self.auth.authenticated:
-            yield self.auth.authenticate()
-        ret = yield self.message_client.send(
-            self._package_load(self.auth.crypticle.dumps(load)),
-            timeout=timeout,
-            tries=tries,
-        )
-        key = self.auth.get_keys()
-        if HAS_M2:
-            aes = key.private_decrypt(ret["key"], RSA.pkcs1_oaep_padding)
-        else:
-            cipher = PKCS1_OAEP.new(key)
-            aes = cipher.decrypt(ret["key"])
-        pcrypt = salt.crypt.Crypticle(self.opts, aes)
-        data = pcrypt.loads(ret[dictkey])
-        data = salt.transport.frame.decode_embedded_strs(data)
-        raise salt.ext.tornado.gen.Return(data)
+    _resolver_configured = False
 
-    @salt.ext.tornado.gen.coroutine
-    def _crypted_transfer(self, load, tries=3, timeout=60):
-        """
-        In case of authentication errors, try to renegotiate authentication
-        and retry the method.
-        Indeed, we can fail too early in case of a master restart during a
-        minion state execution call
-        """
+    @classmethod
+    def _config_resolver(cls, num_threads=10):
+        import salt.ext.tornado.netutil
```

Comment:
**question** another late import - extra, or needs some docs? FWIW, I removed this locally, changed to tcp transport, and didn't appear to have any adverse effects :shrug:

Reply:
Yeah, this was cruft.